src/hooks/open-ai/useAppState/useAppState.tsx

```diff
@@ -0,0 +1,31 @@
+import { createContext, useState, useContext, FC } from 'react'
+
+interface ContextType {
+    openAiToken: string;
+    setOpenAiToken: React.Dispatch<React.SetStateAction<string>>;
+}
+type ContainerProps = {
+    children: React.ReactNode; //👈 children prop typr
+};
+
+export const OpenAiTokenContext = createContext<ContextType | undefined>(undefined);
+
+const AppStateProvider = (props: ContainerProps) => {
+    const [openAiToken, setOpenAiToken] = useState('');
+
+    return (
+        <OpenAiTokenContext.Provider value={{openAiToken, setOpenAiToken}}>
+            {props.children}
+        </OpenAiTokenContext.Provider>
+    )
+}
+
+export { AppStateProvider }
+
+export const useOpenAiToken = () => {
+    const context = useContext(OpenAiTokenContext);
```

Comment:
De-structure this here and pass the piece individually.

I.e
`const {openAiToken, setOpenAiToken} = useContext(/*...*/)`
and 
`return {openAiToken, setOpenAiToken}`

Reply:
Reasoning being that we will inevitably need to do this as the hook grows.